\documentclass[10pt]{beamer}
\usetheme{Warsaw}
\usepackage{graphicx}
\usepackage[utf8]{inputenc}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage[notocbib]{apacite}

\bibliographystyle{apacite}

\setbeamertemplate{caption}{\centering\insertcaption\par}
\setlength{\belowcaptionskip}{15pt}
\renewcommand{\thetable}{}

\providecommand{\e}[1]{\ensuremath{\times 10^{#1}}}

\begin{document}


\date{}
\author{Micha\l{} Burdukiewicz}
\institute{ 
University of Wroc\l{}aw, Department of Genomics, Poland}

\title{MethanoGram and alignment-free phylogeny}


\begin{frame}
\maketitle

<<echo = FALSE,message=FALSE,results='asis'>>=
library(reshape2)
library(ggplot2)
library(grid)
library(biogram)
library(xtable)
library(seqinr)
library(signalHsmm)

theme_presentation <- theme(plot.background=element_rect(fill = "transparent",
                                                   colour = "transparent"),
                      plot.margin = unit(c(1,1,1,1), "cm"),
                      axis.text.x = element_text(size=7), 
                      axis.text.y = element_text(size=7),
                      axis.title.x = element_text(size=11, vjust = -1), 
                      axis.title.y = element_text(size=11, vjust = 1),
                      strip.text = element_text(size=11),
                      legend.text = element_text(size=7), 
                      legend.title = element_text(size=11),
                      plot.title = element_text(size=17),
                      panel.grid.major = element_line(colour="lightgrey", linetype = "dashed"),
                      panel.background = element_rect(fill = "transparent",colour = "black"),
                      legend.background = element_rect(fill="NA"))

@



\end{frame}

\begin{frame}
\frametitle{Outline}
\tableofcontents
\end{frame}

\section{Motivation}


\begin{frame}
MethanoGram is able to relate frequency of n-grams derived from molecular markers to culture conditions.

\medskip

n-gram frequency is reflecting phylogenetic relationships between bacterial strains which are direcly associated with culture conditions.
 \end{frame}

\begin{frame}
Alignment-free, n-gram based methods for sequence comparison:
\begin{itemize}
\item are usually faster, 
\item do not require scoring schemes,
\item \textbf{do not require gene selection}.
\end{itemize}

\end{frame}


\begin{frame}
\textit{More than just a gut feeling: constraint-based genome-scale metabolic models for predicting functions of human intestinal microbes}, Willem M de Vos Laboratory, Microbiome 2017

Input: Genome-scale metabolic models (GEMs).

Results:
\begin{itemize}
\item the prediction of minimal, synthetic, or defined media; 
\item the prediction of possible functions and phenotypes; 
\item the prediction of interspecies interactions.
\end{itemize}

\end{frame}

\begin{frame}{Aim}
Replace GEMs with improved n-gram analysis.

Challanges:
\begin{itemize}
\item more diverse species; 
\item lower quality of data; 
\item longer branches.
\end{itemize}
\end{frame}


\AtBeginSection[]
{
\begin{frame}<beamer>
\frametitle{Outline}
\tableofcontents[currentsection]
\end{frame}
}

\section{Improvements of MethanoGram}

\begin{frame}
Although alignment-free methods can be used to recover topology of a phylogeny, they cannot represent lengths of the branches, which require number of substititutions per site. Despite that, some n-gram based methods can also calculate branch lenghts~\cite{yi_co-phylog:_2013}.
\end{frame}

\subsection{Simple n-gram distance}

\begin{frame}
First usage of n-gram analysis: differences of overlapping and countinous 2- or 3-grams between sets of sequences~\cite{blaisdell_measure_1986}.
\end{frame}

\begin{frame}
We consider only two sequences, $Q$ and $S$, that contain only symbols from the alphabet $u$.

\begin{equation}
d_{ngram} \left( Q, S \right) = \sum_{i = 1}^{u^n} (q_i - s_i)^2
\end{equation}

\begin{itemize}
\item $d_{ngram}$: n-grams distance between sequences;
\item $q_i$: frequency of the i-th of $u^n$ possible substrings of length $n$ in $Q$;
\item $s_i$: frequency of the i-th of $u^n$ possible substrings of length $n$ in $S$;
\end{itemize}

\end{frame}

\begin{frame}
The simple n-gram distance is not powerful when applied to very similar sequences~\cite{hohl_is_2007}.
\end{frame}

\subsection{Frequency corrections}

\begin{frame}

The frequency of words varies between sequences independently of their relationships.

Practical corrections to remove random background:
\begin{itemize}
\item zero-order Markov method substracts background~\cite{pride_evolutionary_2003};
\item composition vector method~\cite{qi_whole_2004};
\item feature frequency profile~\cite{sims_whole-genome_2011}.
\end{itemize}
\end{frame}

\subsection{Composition vector method}

\begin{frame}
The composition vector method substracts random background by normalizing counts of the n-grams~\cite{qi_whole_2004}.

\begin{equation}
d_{cv} \left( Q, S \right) = \frac{1 - C(Q, S)}{2}
\end{equation}

\begin{equation}
C(Q, S) = \frac{\sum{q_{ni} s_{ni}}}{\sqrt{\sum{q_{ni}^2} \sum{s_{ni}^2}}}
\end{equation}

\begin{itemize}
\item $d_{cv}$: composition vector distance;
\item $q_{i}$: normalized frequency of the i-th of $u^n$ possible substrings of length $n$ in $Q$;
\item $s_{i}$: normalized frequency of the i-th of $u^n$ possible substrings of length $n$ in $S$.
\end{itemize}



\end{frame}

\section{Semi-grammar methods}

\begin{frame}

Closely related sequences contain more exact matches than divergent sequences~\cite{kurtz_versatile_2004}.

Differences between semi-grammar methods and n-gram methods:
\begin{itemize}
\item no longer requires specifying $n$;
\item preserves sequence of the substrings;
\item potentially much longer computation time.
\end{itemize}
\end{frame}

\begin{frame}

Suffix tree allows quicker deconstruction of the sequence.

See:
$$S = \textrm{TCCT\$}$$


\end{frame}

\section{Grammar methods}

\begin{frame}

A grammar is a set of rules for decomposing a string into its elements.

$$\textrm{TCCT} \rightarrow \textrm{T}_1 \textrm{C}_2 \textrm{T}_1 $$

Differences between semi-grammar methods and grammar methods:
\begin{itemize}
\item even slower than grammar methods;
\item calculates mutation distances.
\end{itemize}

\end{frame}


\begin{frame}

Common methods of compression: Lempel-Ziv factorization, average common substring.

\medskip

Grammar methods are the msot powerful aligment-free methods~\cite{otu_new_2003}, but are \textbf{more sensitive} to imperfections in data.
\end{frame}



\section{References}

\begin{frame}[allowframebreaks]


\bibliography{kmer_filo}

\end{frame}

\end{document}